File: lib/lambda/rules/aws_cis/aws_cis_3_13.ts
import { cloudTrailRecord, uguisuRule, detection } from "../../models";

// https://docs.aws.amazon.com/securityhub/latest/userguide/securityhub-cis-controls.html#cis-3.13-remediation

export class rule extends uguisuRule {
  constructor() {
    super({
      id: "aws_cis_3.13",
      title: "Route table changes",
      description:
        "AWS CIS benchmark 3.13 recommend to ensure a log metric filter and alarm exist for route table changes",
      severity: "medium",
    });
  }

  detect(record: cloudTrailRecord): detection | null {
    const targetEvents: { [key: string]: boolean } = {
      CreateRoute: true,
      CreateRouteTable: true,
      ReplaceRoute: true,
      ReplaceRouteTableAssociation: true,
      DeleteRouteTable: true,
      DeleteRoute: true,
      DisassociateRouteTable: true,
    };

    if (targetEvents[record.eventName] === true) {
      return {
        rule: this,
        event: record,
      };
    }

    return null;
  }
}

File: pkg/rules/aws_cis_3_13.go
package rules

import (
	"github.com/m-mizutani/uguisu/pkg/models"
)

// https://docs.aws.amazon.com/securityhub/latest/userguide/securityhub-cis-controls.html#cis-3.13-remediation
type awsCIS3_13 struct {
	targetEvents map[string]bool
}

func newAwsCIS3_13() models.Rule {
	return &awsCIS3_13{
		targetEvents: map[string]bool{
			"CreateRoute":                  true,
			"CreateRouteTable":             true,
			"ReplaceRoute":                 true,
			"ReplaceRouteTableAssociation": true,
			"DeleteRouteTable":             true,
			"DeleteRoute":                  true,
			"DisassociateRouteTable":       true,
		},
	}
}

func (x *awsCIS3_13) ID() string                { return "aws_cis_3.13" }
func (x *awsCIS3_13) Title() string             { return "Route table changes" }
func (x *awsCIS3_13) Severity() models.Severity { return models.SeverityMedium }
func (x *awsCIS3_13) Description() string {
	return "AWS CIS benchmark 3.13 recommend to ensure a log metric filter and alarm exist for route table changes"
}
func (x *awsCIS3_13) Match(record *models.CloudTrailRecord) bool {
	return x.targetEvents[record.EventName]
}
